Rakefile

```diff
@@ -40,3 +40,15 @@ RSpec::Core::RakeTask.new(:spec) do |t|
   t.rspec_opts = '--format documentation'
 end
 
+# rake plugins
+task :plugins do
+  puts 'Registering plugins'
+  # Get name of each plugin from folder name. Require each plugin and register.
+  # At this point, we are assuming the plugin is *not* registered and is
+  # therefore not in the plugins table.
+  Dir['plugins/*'].each do |name|
```

Comment:
@ElijahCaine actually, this is the best way to do this because `File.split()` only splits off the last part of the path. If I use `Dir[plugins/*/plugin.rb].each` the path will be `plugins/DiskSize/plugin.rb` for the DiskSize plugin. When I split that I get an array `['plugins/DiskSize', 'plugin.rb']`. I would then need to split the array's zeroeth element again to get the `DiskSize` string.

Basically, using `Dir[plugins/*/plugin.rb].each` means more complexity than using `Dir['plugins/*'].each` and `require "#{name}/plugin.rb"`.